syntax = "proto3";

package gossip;

service Gossip {
  rpc Gossip(GossipRequest) returns (GossipResponse) {}
  rpc History(HistoryRequest) returns (HistoryResponse) {}
}

message GossipRequest {
}

message GossipResponse {
  ValueEntry value_entry = 1;
}

message HistoryRequest {
}

message HistoryResponse {
  repeated ValueEntry value_history = 1;
}

message ValueEntry {
  int32 num_gossip_calls = 1;
  int32 value = 2;
}